scripts/prepare_sector_network.py

```diff
@@ -208,19 +208,33 @@ def add_hydrogen(n, costs):
             x=n.buses.loc[list(nodes)].x.values,
             y=n.buses.loc[list(nodes)].y.values,
         )
-
-        n.madd(
-            "Link",
-            nodes + " H2 Electrolysis",
-            bus0=nodes,
-            bus1=nodes + " grid H2",
-            p_nom_extendable=True,
-            carrier="H2 Electrolysis",
-            efficiency=costs.at["electrolysis", "efficiency"],
-            capital_cost=costs.at["electrolysis", "fixed"],
-            lifetime=costs.at["electrolysis", "lifetime"],
-        )
-
+        if snakemake.config["sector"]["hydrogen"]["water_network"]: #TODO: Adjust bus2 and efficiency2 as needed
+            n.madd(
+                "Link",
+                nodes + " H2 Electrolysis",
+                bus0=nodes,
+                bus1=nodes + " grid H2",
+                bus2=nodes + " H2O",
+                p_nom_extendable=True,
+                carrier="H2 Electrolysis",
+                efficiency=costs.at["electrolysis", "efficiency"],
+                efficiency2=costs.at["desalination", "efficiency"],

```

Comment:
Please use an efficiency describing the ratio of water to hydrogen. If you have it, remember to also consider the efficiency of hydrogen to electricity, so in the end you get the efficiency of water to electricity as the field `efficiency2` intends to incorporate :)

For now, you can fetch the water-hydrogen-efficiency from the config. In the long term, it should/could be from technology-data.